public/modules/custom/helfi_ptv_integration/src/HelfiPTV.php

```diff
@@ -0,0 +1,764 @@
+  private function getPhoneNumbers($phoneNumbers): array {
+    $phoneData = [];
+    foreach ($phoneNumbers as $phoneNumber) {
+      if ($phoneNumber->type !== 'Phone') {
+        continue;
+      }
+      $additional = $phoneNumber->additionalInformation !== null ? $phoneNumber->additionalInformation . ' ' : '';
+      $phoneData[$phoneNumber->language][] = $additional . $phoneNumber->prefixNumber . $phoneNumber->number;
+    }
+    return $phoneData;
+  }
+
+  /**
+   * @param $emailAddresses
+   * @return array
+   */
+  private function getEmails($emailAddresses): array {
+    $emailData = [];
+    foreach ($emailAddresses as $emailAddress) {
+      $emailData[$emailAddress->language][] = $emailAddress->value;
+    }
+    return $emailData;
+  }
+
+  /**
+   * @param $cityId
+   * @param $date
+   * @return mixed
+   */
+  private function getOrganizations($cityId) {
+    if ($cityId == 'all') {
+      $terms = $this->getMunicipalityTerms();
+      $bodyData = [];
+      foreach ($terms as $term) {
+        if ($term->field_ptv_code->value != '') {
+          $params = [
+            'query' => [
+              'includeWholeCountry' => 'true',
+              'showHeader' => 'true',
+            ]
+          ];
+          try {
+            $response = $this->httpClient->get(
+              'https://api.palvelutietovaranto.suomi.fi/api/v11/Organization/area/Municipality/code/' . $term->field_ptv_code->value,
+              $params
+            );
+          } catch (ClientException $e) {
+            $this->loggerFactory->notice('Getting organizations failed for city ID ' . $term->field_ptv_code->value);
+            $this->loggerFactory->notice($e->getMessage());
+            continue;
+          } catch (GuzzleException $e) {
+            $this->loggerFactory->notice('Getting organizations failed for city ID ' . $term->field_ptv_code->value);
+            $this->loggerFactory->notice($e->getMessage());
+            continue;
+          } catch (\Exception $exception) {
+            $this->loggerFactory->notice('Getting organizations failed for city ID ' . $term->field_ptv_code->value);
+            $this->loggerFactory->notice($exception->getMessage());
+            continue;
+          }
+
+          if ($response->getStatusCode() !== 200) {
+            $this->loggerFactory->notice('Getting organizations failed for city ID ' . $term->field_ptv_code->value);
+            continue;
+          }
+          $body = JSON::decode($response->getBody());
+          foreach ($body['itemList'] as $item) {
+            $bodyData[] = $item;
+          }
+          $pages = $body['pageCount'];
+          if ($pages > 1) {
+            for ($p = 1; $p <= $pages; $p++) {
+              $params['query']['page'] = $p;
+              try {
+                $response = $this->httpClient->get(
+                  'https://api.palvelutietovaranto.suomi.fi/api/v11/Organization/area/Municipality/code/' . $term->field_ptv_code->value,
+                  $params
```

Comment:
Could all these try catch calls to `https://api.palvelutietovaranto.suomi.fi/api/v11/Organization/area/Municipality/code/` be added as separate function because those are almost similar and It would make the code easier to read.